agent/metrics/metrics.go

```diff
@@ -0,0 +1,214 @@
+/*
+Copyright 2019 Google Inc. All rights reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+        "context"
+        "fmt"
+        "log"
+        "time"
+
+	apioption "google.golang.org/api/option"
+	gax "github.com/googleapis/gax-go/v2"
+        googlepb "github.com/golang/protobuf/ptypes/timestamp"
+        metricpb "google.golang.org/genproto/googleapis/api/metric"
+        monitoring "cloud.google.com/go/monitoring/apiv3"
+        monitoringpb "google.golang.org/genproto/googleapis/monitoring/v3"
+        monitoredrespb "google.golang.org/genproto/googleapis/api/monitoredres"
+)
+
+const (
+	samplePeriod	= 60 * time.Second
+        endpoint	= "monitoring.sandbox.googleapis.com:443"
+)
+
+var startTime time.Time
+var codeCount map[string]int64
+
+// metricClient is a client for interacting with Cloud Monitoring API.
+type metricClient interface {
+	CreateTimeSeries(ctx context.Context, req *monitoringpb.CreateTimeSeriesRequest, opts ...gax.CallOption) error
+}
+
+// fakeMetricClient is a stub of MetricClient for the purpose of testing
```

Comment:
This should be moved into the `metrics_test.go` file instead of being here.

Reply:
Done.